lib/pure/strutils.nim

```diff
@@ -979,6 +992,46 @@ proc parseHexInt*(s: string): int {.noSideEffect, procvar,
     of '\0': break
     else: raise newException(ValueError, "invalid integer: " & s)
 
+proc generateHexCharToValueMap(): string =
+  ## Generate a string to map a hex digit to uint value
+  result = ""
+  for inp in 0..255:
+    let ch = chr(inp)
+    let o =
+      case ch:
+        of '0'..'9': ord('0')
+        of 'a'..'f': ord('a') - 10
+        of 'A'..'F': ord('A') - 10
+        else: -1
+    if o == -1:
```

Comment:
if you do not split it into 'case + if' the code becomes simpler.

Reply:
Updated.